core/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java

```diff
@@ -333,6 +343,45 @@ public long getTotalTimeMS(int pointIndex) {
         return getTotalTimesMS()[pointIndex];
     }
 
+    /** Returns the total energy consumed at train wheel on this envelopePart */
+    public double getEnergyConsumed(
```

Comment:
same as above, I know it's less convenient, but this stuff should really be kept out of the `envelope` module, as it does not have to be here: it can just use the public API

Reply:
Done. We moved it to `EnvelopePhysics.java`